app/src/lib/components/PowerTable.svelte

```diff
@@ -725,10 +725,11 @@ function handleSubmittedEdits(event: CustomEvent){
     let rowIndex: number = event.detail.rowIndex;
     let row = data[pageData[rowIndex][dataIdKey]];
     let rowInputs = (<HTMLInputElement>event.detail.domEvent.target).closest('tr')?.querySelectorAll(`[data-name=edit-input]`);
-    
-    if (!rowInputs) {
-        // ToDo: error should be easily visible to the end user
-        console.log('error');
+
+    const numRenderedColumns = instructs.length - 1 // minus one for checkbox column
+    if (rowInputs.length !== numRenderedColumns) {
```

Comment:
Typescript complains that `rowInputs` might be undefined.

This will result in the cell value to be replaced with an empty string (once an edit button is clicked):

```svelte
<div data-key={instructKey} data-name='edit-input'>
	{value}
</div>
```

and this will result in a new key (an empty string) to be added to the row:

```svelte
<div data-name='edit-input'>
	{value}
</div>
```

It would be better not to require `data-name='edit-input'` on an element like div. 

I haven't thought about the store idea enough to know exactly how it should be done, but if you end up implementing it for yourself and decide to share your code via a PR it would be appreciated.

Reply:
In regards to Typescript complaining, what do you think is the best way to handle the warning? The options I see are
- adding a `// @ts-ignore`, which is my initial preference
- return when `rowInputs` is `null`, but that should never really happen, and if it does, you got bigger worries

At the fear of wasting your time, and being too pedantic for a simple change, I've put my additional mulling behind a paywall as a deterrent.
<details><summary>$5 paywall</summary>

From what I understand, the warning is occurring because in the line
```typescript
let rowInputs = (<HTMLInputElement>event.detail.domEvent.target).closest('tr')?.querySelectorAll(`[data-name=edit-input]`);
```
 [`.closest('tr')` could return null](https://developer.mozilla.org/en-US/docs/Web/API/Element/closest#return_value) (and note that `.querySelectorAll()` always [returns an array](https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelectorAll#return_value)). If it is the case that `rowInputs` is null, then the page somehow has no table rows (even a filtered table with no results still will have a `<tr>`). The check below can get rid of the warning, but probably isn't too useful to tell a user they don't have a table:
```typescript
    if (!rowInputs) {
        alert('No table rows found');
        return;
    }
```

Unless I am misunderstanding the error and behaviors, I think adding a `// @ts-ignore` here is most appropriate? Adding the null check code above fixes the issue, but would purely be for satisfying Typescript, and not the users. Let me know if I understand correctly, and which option you prefer me to add.
</details>

---

In regards to the non-editable div example, I am not sure how to get around the issues you mention. That is why I didn't end up adding such an example into the example7/ dir, as that would indicate official support. Apologies that I didn't communicate my thoughts there previously.

One easy way to solve the issues, but would remove the validation check you requested previously, is to remove the `if (rowInputs.length !== numRenderedColumns)` check. Then the non-editable example HTML could become
```
<div>
	<p>{value}</p>
</div>
```

Are ok with removing that check? Or maybe there is a more clever check to validate the user's edit-component has the necessary `data-name='edit-input`, but I haven't thought of one. An aside is below if you care to discuss in more depth why I think it is ok to remove the check.
<details><summary>aside</summary>

In my opinion, it shouldn't be up to the library at runtime to worry about how a client implemented a custom component. Since we don't have "compile time" ways to do these checks, or enforce an interface or template, the responsibility should fall on the library's documentation being good, and the developer reading said documentation to make sure they implemented things correctly. The only benefit of having a runtime error, that I can quickly think of, is that it makes bugs more triage-able.
</details>

Reply:
In this case `// @ts-ignore` makes sense, but is not ideal (it disables the warnings indiscriminately, you have to use it at least for two lines, and it wouldn't be clear why it's there).

Instead you can either change `.closest('tr')?` to `.closest('tr')!` to assert that there is going to be a `tr` tag, or you can add the null check to the same line you're checking the number of rendered columns:

```js
const numRenderedColumns = instructs.length - 1; // minus one for checkbox column
if (!rowInputs || rowInputs.length !== numRenderedColumns) {
    alert(`Only found ${rowInputs?.length} of ${numRenderedColumns} expected inputs. Make sure 'data-name=edit-input' attribute is on your component's input field`);
    console.error('Misconfigured edit component inputs');
    return;
}
```
<sub>Side note: I have also added `;` to the end of the first line.</sup>

You can remove the checks, but ideally we want to make sure to somehow handle those inputs that don't have the proper attributes (e.g. data-name). Otherwise user might edit the content and submit the changes without noticing that some of the changes were ignored. Developers should follow the documentations, but mistakes happen and we can at least try to make sure they won't go unnoticed whenever possible. 

```js
row[(<HTMLInputElement>rowInput)?.dataset?.key ?? ''] = (<HTMLInputElement>rowInput)?.value ?? '';
```

If `key` or `value` return null we don't want to substitute them with `''` and move on. So, this might be a good place to check and see if displaying an error is useful. Since `handleSubmittedEdits` is performed on a single row, we can afford to iterate through the cells with more flexibility. We can also use CSS to disable common form control elements of an edit block that aren't explicitly identified as `edit-input`, etc.

Reply:
I liked your idea for checking `key` and `value`. Just pushed the changes for it.